publications/tasks.py

```diff
@@ -7,228 +7,239 @@
-            if identifier_value and Publication.objects.filter(url=identifier_value).exists():
-                logger.info("Skipping duplicate publication (URL): %s", identifier_value)
+            if identifier and Publication.objects.filter(url=identifier).exists():
                 continue
-            # Skip records without a valid URL.
-            if not identifier_value or not identifier_value.startswith("http"):
-                logger.warning("Skipping record with invalid URL: %s", identifier_value)
+            if not identifier or not identifier.startswith("http"):
                 continue
 
-            geom_object = GeometryCollection()
-            period_start, period_end = [], []
+            geom = GeometryCollection()
+            ps_list: list[str] = []
+            pe_list: list[str] = []
+    # … earlier in parse_oai_xml_and_save_publications …
             try:
-                resp = requests.get(identifier_value, timeout=10)
+                resp = requests.get(identifier, timeout=10)
                 resp.raise_for_status()
                 soup = BeautifulSoup(resp.content, "html.parser")
-
-                try:
-                    geom = extract_geometry_from_html(soup)
-                    geom_object = geom or GeometryCollection()
-                except Exception as geo_err:
-                    logger.error("Geometry extraction failed for URL %s: %s", identifier_value, geo_err)
-                    geom_object = GeometryCollection()
-
-                try:
-                    start_time, end_time = extract_timeperiod_from_html(soup)
-                    if isinstance(start_time, list):
-                        period_start = [d for d in start_time if d]
-                    if isinstance(end_time, list):
-                        period_end = [d for d in end_time if d]
-                except Exception as time_err:
-                    logger.error("Time period extraction failed for URL %s: %s", identifier_value, time_err)
-
-            except Exception as fetch_err:
-                logger.error("Error fetching HTML for %s: %s", identifier_value, fetch_err)
-                geom_object = GeometryCollection()
-                period_start, period_end = [], []
-
-            publication = Publication(
-                title=title_value,
-                abstract=abstract_text,
-                publicationDate=date_value,
-                url=identifier_value,
-                doi=doi_text,
-                source=journal_value,
-                geometry=geom_object,
-                timeperiod_startdate=period_start,
-                timeperiod_enddate=period_end,
-                job=event
+                g = extract_geometry_from_html(soup)
+                if g:
+                    geom = g
+            
+                if src and getattr(src, "is_preprint", False) and geom.empty:
+                    try:
+                        loc = Nominatim(user_agent="optimap-tasks").geocode(
```

Comment:
Where does the geocoding come from? Can you please clarify which issue this implements?